msu/classes/weighted_container.nut

```diff
@@ -0,0 +1,93 @@
+::MSU.Class.WeightedContainer <- class
+{
+	Total = null;
+	Array = null;
+
+	constructor( _array = null )
+	{
+		if (_array == null) _array = [];
+		::MSU.requireArray(_array);
+		this.Total = 0;
+		this.Array = [];
+		this.extend(_array);
+	}
+
+	function extend( _array )
+	{
+		::MSU.requireArray(_array);
+		foreach (i in _array)
+		{
+			this.push(i);
+		}
+	}
+
+	function push( _item )
+	{
+		if (typeof _item != "array") _item = [1, _item];
+		else
+		{
+			if (_item.len() != 2)
```

Comment:
This can work with floats as well so I'll do a requireOneOfType

Reply:
I don't think we should allow float type weights. It makes no sense from a practicality point of view.

Reply:
Not sure what you mean? Normally you'd wanna stick to ints but there's not practical reason you have to. For example say you have something set up and you want to add a very low chance of something else dropping, you could just add that new thing with a weight of 0.01 instead of having to redo everything